### DISCUSSION TITLE
Syslog timestamp mismatch

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#8182] [Unsupported Versions] [answered]
Hello!

I successfully import Firewall syslogs (including `syslog.timestamp`), but now I saw that the ingest timestamp does not match the syslog timestamp. I found the following two lines in the SO configuration of the syslog ingest pipeline that I use (with custom GROK patterns in `/opt/so/saltstack/local/salt/elasticsearch/files/ingest`, I did not change the lines below):

```
{ "set": { "field": "ingest.timestamp",      "value": "{{ @timestamp }}"       } },
{ "date": { "if": "ctx.syslog?.timestamp != null", "field": "syslog.timestamp",       "target_field": "@timestamp",   "formats": ["MMM  d HH:mm:ss", "MMM dd HH:mm:ss", "ISO8601", "UNIX"], "ignore_failure": true  } }, 
```
Aren't they in the wrong order? Don't I have to extract my syslog.timestamp to @timestamp first before I try to write @timestamp to ingest.timestamp?
 
BTW: What is `ctx.syslog?.timestamp`? Doesn't this has to be `syslog.timestamp`?

Cheers, Ben

**Comment by weslambert** [comment, marked as answer]
`ingest.timestamp`  refers to when the log was ingested. 
`syslog.timestamp` refers to the time when the syslog message was generated.

`@timestamp` is created automatically for the event by Elasticsearch, but we don't want to use that when searching for the event, because an analyst is more likely to  be searching based on the syslog event time, not the ingestion time.

**Reply by ben-sec** [reply]
I guess the problem is, that you expect `syslog.timestamp` to be UTC, correct? But in my case it's CET. I will try to add `"timezone": "Europe/Berlin"`:

```
{ "date": { "if": "ctx.syslog?.timestamp != null", "field": "syslog.timestamp",       "target_field": "@timestamp",   "formats": ["MMM  d HH:mm:ss", "MMM dd HH:mm:ss", "ISO8601", "UNIX"], "timezone": "Europe/Berlin",   "ignore_failure": true  } }
```

According to this [blog](https://alexmarquardt.com/2019/10/16/converting-local-time-to-iso-8601-time-in-elasticsearch/) it should help.

**Reply by ben-sec** [reply]
Setting the correct time zone fixed it:

Before:
![image](https://user-images.githubusercontent.com/81571847/176184176-a470c56c-9015-4d99-b4c3-3a092bed625c.png)

After:
![image](https://user-images.githubusercontent.com/81571847/176184331-e6fa21de-c224-4d57-a6bd-286593884d83.png)
